File: uploadservice/src/main/java/net/gotev/uploadservice/DefaultLoggerDelegate.java
package net.gotev.uploadservice;

import android.util.Log;

/**
 * Default logger delegate implementation which logs in LogCat with {@link Log}.
 * Log tag is set to <b>UploadService</b> for all the logs.
 * @author gotev (Aleksandar Gotev)
 */
public class DefaultLoggerDelegate implements Logger.LoggerDelegate {

    private static final String TAG = "UploadService";

    @Override
    public void error(String tag, String message) {
        Log.e(TAG, tag + " - " + message);
    }

    @Override
    public void error(String tag, String message, Throwable exception) {
        Log.e(TAG, tag + " - " + message, exception);
    }

    @Override
    public void debug(String tag, String message) {
        Log.d(TAG, tag + " - " + message);
    }

    @Override
    public void info(String tag, String message) {
        Log.i(TAG, tag + " - " + message);
    }
}

File: uploadservice/src/main/java/net/gotev/uploadservice/logger/DefaultLoggerDelegate.kt
package net.gotev.uploadservice.logger

import android.util.Log

class DefaultLoggerDelegate : UploadServiceLogger.Delegate {

    companion object {
        private const val TAG = "UploadService"
    }

    override fun error(component: String, uploadId: String, message: String, exception: Throwable?) {
        Log.e(TAG, "$component - (uploadId: $uploadId) - $message", exception)
    }

    override fun debug(component: String, uploadId: String, message: String) {
        Log.i(TAG, "$component - (uploadId: $uploadId) - $message")
    }

    override fun info(component: String, uploadId: String, message: String) {
        Log.i(TAG, "$component - (uploadId: $uploadId) - $message")
    }
}
